com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/TextureGenerator.java

```diff
@@ -259,14 +259,28 @@ private static ITextureCompressor getDefaultTextureCompressor() {
 
     private static List<Long> GenerateImages(long image, int width, int height, boolean generateMipChain) throws TextureGeneratorException {
         List<Long> images = new ArrayList<>();
-        int mipWidth = width;
-        int mipHeight = height;
-        int mipLevel = 0;
-        long prevImage = image;
-
-        while (mipWidth != 0 || mipHeight != 0) {
-            mipWidth = Math.max(mipWidth, 1);
-            mipHeight = Math.max(mipHeight, 1);
+        int baseWidth = TexcLibraryJni.GetWidth(image);
+        int baseHeight = TexcLibraryJni.GetHeight(image);
+        boolean baseMatches = baseWidth == width && baseHeight == height;
+
+        // Use the provided image as mip0 if it matches the requested size; otherwise resize once to seed the chain.
```

Comment:
if provided image == mip0, use it instead of calling `TexcLibraryJni.Resize` which makes buffer copy and for tested image it takes about 100ms